### PR TITLE
Defile Ability Rework

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

1. Defile wind up reduced from 1.5 seconds to 0.5 seconds.
2. Defile plasma cost reduced from 150 to 100.
4. Defile Brute damage reduced from 10 to 5 and Stamina damage reduced from 100 to 50.
4. Defile no longer injects larva; instead it purges all chemicals in the victim and deals toxin damage equal to half the total U of chemical purged from the victim's system multiplied by 2 for each unique xeno toxin in its system. The victim also exudes a cloud of blood red Sanguinal smoke that scales in size with the amount of toxin damage dealt in this way.
5. Sanguinal is a new toxin that deals minor brute and bleeding each tick.  For each other unique xeno toxin in the victim's system, it will deal minor damage of a corresponding type (Stamina for Hemodile, Toxin for Neurotoxin and Burn for Transvitox).

## Why It's Good For The Game

Removes vestiges of 'capture gameplay'.

Adds interesting and synergistic 'finisher' move to the Defiler that combos with its other abilities and other xenos.

## Changelog
:cl:
balance: Defile wind up reduced from 1.5 seconds to 0.5 seconds.
balance: Defile plasma cost reduced from 150 to 100.
balance: Defile Brute damage reduced from 10 to 5 and Stamina damage reduced from 100 to 50.
add: Defile no longer injects larva; instead it purges all chemicals in the victim and deals toxin damage equal to half the total U of chemical purged from the victim's system multiplied by 2 for each unique xeno toxin in its system. The victim also exudes a cloud of blood red Sanguinal smoke that scales in size with the amount of toxin damage dealt in this way.
add: New Sanguinal toxin added; deals minor brute and bleeding each tick.  For each other unique xeno toxin in the victim's system, it will also deal minor damage of a corresponding type (Stamina for Hemodile, Toxin for Neurotoxin and Burn for Transvitox).
/:cl:

### CODE REVIEW

**Comment by Lefunny** [comment]
this sounds funny but we still need more balancment stuff for defiler, maybe deal some of his """super strong""" slash for speed or armor as it is a caste that need to deal with close range but dont have enough speed or armor to have a "safe" gameplay.

**Comment by bobmancer343** [comment]
I'd still want some larva defiler stings maybe as a seperate ability or a switchable one. Dunno I just like being a dickhead and defiling crit marines with larva.

**Review by BraveMole — CHANGES_REQUESTED** [review]
I like the concept, but i'm not a fan of the cd, plasma cost and windup reduction.

If it's a finisher, it doesn't need low plasma cost or low cd?

The windup is maybe fine on its own, but with the rest it's just too much

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 1]
```diff
@@ -2,45 +2,94 @@
...
+		if(is_type_in_typecache(current_reagent, GLOB.defile_purge_list)) //For each xeno toxin reagent, double the strength multiplire
+			defile_strength_multiplier *= 2
+		living_target.reagents.remove_reagent(current_reagent.type,defile_reagent_amount) //Purge current chem
+
+	defile_power = DEFILER_DEFILE_BASE_POWER + defile_reagent_amount * defile_strength_multiplier //Total amount of toxin damage we deal
+
+	living_target.setToxLoss(min(200, living_target.getToxLoss() + defile_power)) //Apply the toxin damage; cap toxin damage at lower of 200 or defile power + current tox loss
+
+	var/datum/effect_system/smoke_spread/xeno/sanguinal/blood_smoke = new(living_target) //Set up Sanguinal smoke
+	var/turf/target_turf = get_turf(living_target)
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:80]: usless var

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 2]
```diff
@@ -2,45 +2,94 @@
...
+
+	if(!owner.Adjacent(A))
+		var/mob/living/carbon/xenomorph/X = owner
+		if(!silent)
+			to_chat(X, span_warning("We can't reach this target! We need to be adjacent!"))
+		return FALSE
+
+
+/datum/action/xeno_action/activable/defile/use_ability(atom/A)
 	var/mob/living/carbon/xenomorph/Defiler/X = owner
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:46]: at any point, do you need it to be a defiler? If not, just cast as a xeno, this allows action to be portable

**Review by BraveMole — APPROVED** [review]
Unless splinter come and veto, fix this and i'll tm

**Comment by SplinterGP** [comment]
pretty much bonkers with current rainbow slashing, theres #8051 for that however, plus i dunno gotta test the explosion potential more with time for this, just TM for now

**Pull request merged**